thavalon-webapp/src/components/gameComponents/interactions/missionManager.tsx

```diff
@@ -0,0 +1,153 @@
+import { PlayerCard } from "../playerCard";
+
+/**
+ * The required properties for the MissionManager
+ */
+interface MissionManagerProps extends InteractionProps {
+    me: string,
+    message: MissionGoingMessage,
+    votes: Map<string, Vote>,
+}
+
+/**
+ * The required properties for the MissionCard
+ */
+interface MissionCardProps {
+    submitMissionCard(card: MissionCard): void
+}
+
+/**
+ * Required properties for the MissionResult modal.
+ */
+interface MissionResultModalProps {
+    message: MissionResultsMessage | undefined,
+    setOpen: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+/**
+ * Component managing all of the mission related interactions.
+ * @param props Required properties for the MissionManager
+ */
+export function MissionManager(props: MissionManagerProps): JSX.Element {
+    useEffect(() => {
+        const connection = GameSocket.getInstance();
+        connection.onGameEvent.subscribe(handleMessage);
+        return () => connection.onGameEvent.unsubscribe(handleMessage);
+    });
+
+    /**
+     * Handles any incoming server message
+     * @param message A message from the server
+     */
+    function handleMessage(message: InboundMessage): void {
+        // TODO: Support cardPlayed message types here.
+        if (message.messageType !== InboundMessageType.GameMessage) {
+            return;
+        }
+    }
+
+    // State maintaining if the player has played a card or not
+    const [hasPlayedCard, setHasPlayedCard] = useState(false);
+
+    /**
+     * Submits a mission card or QB to the server and updates if the player
+     * has played a card accordingly.
+     * @param card A mission card to play
+     */
+    function submitMissionCard(card: MissionCard): void {
+        if (card === MissionCard.QuestingBeast) {
+            sendGameAction(GameActionType.QuestingBeast);
+        } else {
+            sendGameAction(GameActionType.Play, { card: card });
+            setHasPlayedCard(true);
+        }
+    }
+
+    // Create player cards here
+    const playerCards = props.playerList.map((playerName) => {
+        const selectedTypes = createSelectedPlayerTypesList(playerName, props.primarySelectedPlayers, props.secondarySelectedPlayers);
+        return <PlayerCard
+            key={playerName}
+            name={playerName}
+            vote={props.votes.get(playerName)}
+            tabbedOut={props.tabbedOutPlayers.has(playerName)}
+            selectedTypes={selectedTypes}
+            enabled={false} />
+    });
+
+    const playersOnMission = new Set(props.message.players);
+    return (
+        <>
+            {playerCards}
+            <div className="mission-manager">
+                {!hasPlayedCard && playersOnMission.has(props.me) &&
+                    <MissionCardButtons submitMissionCard={submitMissionCard} />}
+                {(hasPlayedCard || !playersOnMission.has(props.me)) &&
+                    // TODO: Support mission card progress bar here.
```

Comment:
github issue?

Reply:
Already fixed with the playerCard change